src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private static final String PATTERN = "dd.MM.yyyy";
+    private SimpleDateFormat dateFormat = new SimpleDateFormat(PATTERN);
+    private Date startWork;
+    private Date finishWork;
+    private Date dayWork;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        try {
+            startWork = dateFormat.parse(dateFrom);
+            finishWork = dateFormat.parse(dateTo);
+        } catch (ParseException pe) {
+            return "exception-wrecker" + pe;
+        }
+        String[] returnStrings = new String[names.length];
+        int[] sums = new int[names.length];
+        for (int a = 0; a < names.length; a++) {
+            String nameAtIndex = names[a];
+            sums[a] = 0;
+            for (int b = 0; b < data.length; b++) {
+                String dataString = data[b];
+                if (dataString.contains(nameAtIndex)) {
+                    String[] someData = dataString.split(" ");
+                    try {
+                        dayWork = dateFormat.parse(someData[0]);
+                    } catch (ParseException pe) {
+                        return "exception-wrecker" + pe;
+                    }
+                    int ret1 = dayWork.compareTo(startWork);
```

Comment:
do not use such naming(dataString , ret1, ret2, c, d, returnStrings, output1, output2, outputStringBuilder, outputAnswer), how to guess what it stores?